### PR TITLE
Prevent saving the post before previewing in locked/read-only mode.

### LINKED ISSUE #27568: Preview button in edit lock modal is saving the post
**Describe the bug**
If User A is editing a draft post and User B clicks to edit it, they get a modal that the post is being edited, with options to Exit, Preview, or Take Over. (note that Preview is currently hidden on desktop view, but visible in mobile view).
If User B clicks the Preview button, it saves the post and then loads the preview. If metaboxes are present, it actually overrides the edit lock and kicks out User A.

**To reproduce**
1. Add a metabox to the post edit screen.
2. Create a draft post and save it (don't publish)
3. Log in as a different user in a different browser or incognito and edit the same post
4. When you get the modal, show the Preview button if not visible, and click Preview. You will see in the network tab a post to the rest api, and then a post to post.php for the metabox.
5. After the next heartbeat post to admin-ajax.php the original user will get kicked out of the post.

**Expected behavior**
When the second user clicks the Preview button, there should not be a save. It should go to the preview url.

**Screenshots**

**Editor version (please complete the following information):**
- WordPress version: 5.5.3
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? default

**Desktop (please complete the following information):**
 - OS: Mac OS 10.14.6
 - Browser: chrome
 - Version: 86.0.4240.198

**Additional context**
The problem appears to be happening here: https://github.com/WordPress/gutenberg/blob/5cd71f42499173e6d6f10a20a4b066f3aab77e34/packages/editor/src/components/post-preview-button/index.js#L162-L175
The `isAutosaveable` prop does not appear to be checking the edit lock.

This code snippet will add a metabox and show the Preview button:
```
function wporg_add_custom_box() {
	add_meta_box(
		'wporg_box_id',
		'Custom Meta Box Title',
		'wporg_custom_box_html',
		'post'
	);
}
add_action( 'add_meta_boxes', 'wporg_add_custom_box' );
function wporg_custom_box_html( $post ) {
    ?>
    <label for="wporg_field">Description for this field</label>
    <select name="wporg_field" id="wporg_field" class="postbox">
        <option value="">Select something...</option>
        <option value="something">Something</option>
        <option value="else">Else</option>
    </select>
    <?php
}
add_action( 'enqueue_block_editor_assets', function() {
    $style = "
    .editor-post-locked-modal__buttons >.editor-post-preview {
        display: inline-flex;
    }";
    wp_add_inline_style( 'wp-block-editor', $style );
} );
```

### PR DESCRIPTION
**Pull request by desaiuditd**
## Description
#27568 reported that when a post is opened in read-only/locked mode, user is shown a dialog with `Preview` or `Take over` options.
image

If the post is still in draft, and if the new user clicks `Preview`, block editor triggers an autosave request, which basically overrides the draft changes made by the original user who currently holds the post lock.

## Steps to Reproduce
- Login as User A.
- Create a post in block editor. Let's say with Post ID - 1. Do not publish it.
- Login as User B in different browser or incognito mode.
- Open the Post with ID - 1 as User B.
- Observe that the post is locked and you see the dialog.
- Meanwhile, go back to User A window and start making new changes into the post. Save changes.
- Come back to User B window. Click preview.
- Notice from dev console, that save request is triggered.
- Notice in the preview, that you are not seeing new changes made by User A.
- Go back to User A window. Refresh the page and notice that you just lost content.

## How has this been tested?
With the fix in this PR, I tested above reproduction steps and User A is not loosing the content anymore.

## Types of changes
Fixes #27568

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/editor/src/components/post-preview-button/index.js [hunk 1]
```diff
@@ -159,9 +159,14 @@ export class PostPreviewButton extends Component {
...
-		// If we don't need to autosave the post before previewing, then we simply
-		// load the Preview URL in the Preview tab.
-		if ( ! this.props.isAutosaveable ) {
+		if (
+			// If we don't need to autosave the post before previewing, then we simply
+			// load the Preview URL in the Preview tab.
+			! this.props.isAutosaveable ||
+			// Also, save the post only if user has the post lock.
+			// Do not save or overwrite the post, if the post is already locked.
+			this.props.isPostLocked
```
Comment (desaiuditd) [inline packages/editor/src/components/post-preview-button/index.js:173]: Note: @joemcgill suggested to expose this boolean as a component prop to `PostPreviewButton` but in my opinion, we should never perform any create/update changes to a post which is already locked by someone else. So this should be checked in all the cases. So there's no point exposing it as a prop.

File: packages/editor/src/components/post-preview-button/index.js [hunk 2]
```diff
@@ -165,9 +165,14 @@ export class PostPreviewButton extends Component {
...
 		this.previewWindow.focus();
 
-		// If we don't need to autosave the post before previewing, then we simply
-		// load the Preview URL in the Preview tab.
-		if ( ! this.props.isAutosaveable ) {
+		if (
+			// If we don't need to autosave the post before previewing, then we simply
+			// load the Preview URL in the Preview tab.
+			! this.props.isAutosaveable ||
+			// Also, save the post only if user has the post lock.
```
Comment (ntsekouras) [inline packages/editor/src/components/post-preview-button/index.js]: I guess this comment `// Also, save the post only if user has the post lock.` can be removed IMO

**Review by ntsekouras — APPROVED** [review]
LGTM - thanks for your work here!

I've left just a suggestion for comment removal but it good to land.

File: packages/editor/src/components/post-preview-button/index.js (hunk 2 above)
Comment (desaiuditd) [reply packages/editor/src/components/post-preview-button/index.js]: Yeah. 😄 I later realised, both comments (the one below) talks about the same thing. Removed. 40746ad

**Pull request merged**